src/core/logical_layers/utils/layersOrder/layersOrder.ts

```diff
@@ -43,76 +63,120 @@ export class LayersOrderManager {
+
+    const mountedLayersOfGivenType = orderedLayers.get(type);
+
+    const givenLayerParentKey = this._layersParentsIds?.get(id);
+    const givenLayerSettings = this._layersSettings?.get(givenLayerParentKey || '')?.data;
+    if (!givenLayerSettings) {
+      console.warn('settings were not found for layer ', type, id);
+      return;
+    }
+    const givenCategory = givenLayerSettings.category;
+    // since layers are ordered .find method will return first layer to mount underneath
+    const higherLayerOfGivenType = mountedLayersOfGivenType?.find((searchingLayer) => {
+      // for 'base' category return first layer that's not base (overlay and no category)
+      if (givenCategory === 'base') return searchingLayer.category !== 'base';
+      // for overlay return only layers with no category
+      if (givenCategory === 'overlay') return searchingLayer.category === undefined;
+    });
+
+    if (givenCategory && higherLayerOfGivenType) {
+      return higherLayerOfGivenType.layer.id;
+    }
+
+    if (!higherLayerOfGivenType || !givenCategory) {
+      const higherMountedType = this._typesOrder
+        // get all higher types
+        .slice(this._typesOrder.indexOf(type) + 1)
+        // find closest of them that exists on map
+        .find((higherType) => orderedLayers.get(higherType)?.length);
+
+      // Mount on top of the map
+      if (!higherMountedType) return undefined;
+      const higherMountedLayer = orderedLayers.get(higherMountedType)?.[0].layer.id;
+      return higherMountedLayer;
+    }
   }
 
-  _getIdToMountOnTypesBottomSync(
+  private _getIdToMountOnTypesBottomSync(
     map: MapLibre,
     type: LayersType,
+    id: string,
   ): string | undefined {
-    const bottomLayersOfMountedType = this._getMountedBottomLayersByType(
-      map,
-      type,
-    );
-    if (!bottomLayersOfMountedType) return;
-    const currentTypeBottomLayer = bottomLayersOfMountedType.get(type);
-    // if there is bottom layer for the current type - return it to mount underneath it
-    if (currentTypeBottomLayer?.id) return currentTypeBottomLayer.id;
-    // if no layers for current type mounted -
-    // search for the closest top type to mount right underneath it
-
-    const firstHigherMountedType = this._typesOrder
-      // get all higher types
-      .slice(this._typesOrder.indexOf(type) + 1)
-      // find closest of them that exists on map
-      .find((t) => bottomLayersOfMountedType.has(t));
-
-    // if there's no higher type mounted - return undefined and mount layer on top
-    if (!firstHigherMountedType) return;
-    // if higher type is mounted, get it's bottom layer's id to mount underneath it
-    return bottomLayersOfMountedType.get(firstHigherMountedType)?.id;
+    const orderedLayers = this._getMountedOrderedLayers(map);
+    if (!orderedLayers?.size) return;
+    const mountedLayersOfGivenType = orderedLayers.get(type);
+    const givenLayerParentKey = this._layersParentsIds?.get(id);
```

Comment:
And why store this id card in the atom, it's not like we're in any way reactive to it